lib/internal/util.js

```diff
@@ -146,7 +146,13 @@ function cachedResult(fn) {
 //   B() instanceof B // true
 function createClassWrapper(type) {
   function fn(...args) {
-    return Reflect.construct(type, args, new.target || type);
+    const res = Reflect.construct(type, args, new.target || type);
+
+    if (this) {
```

Comment:
`if (this && !new.target)`? Otherwise this would also set the prototype for cases like `new Class()`, right?